apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+
+        _changeCollateralRequirement(disputable, disputableInfo, _collateralToken, _actionAmount, _challengeAmount, _challengeDuration);
+        if (disputable.getAgreement() != IAgreement(this)) {
+            disputable.setAgreement(IAgreement(this));
+        }
+    }
+
+    /**
+    * @notice Deregister `_disputable`
+    * @param _disputableAddress of the disputable app to be unregistered
+    */
+    function unregister(address _disputableAddress) external auth(MANAGE_DISPUTABLE_ROLE) {
+        DisputableInfo storage disputableInfo = disputableInfos[_disputableAddress];
+        _ensureRegisteredDisputable(disputableInfo);
+
+        disputableInfo.registered = false;
+        emit DisputableAppUnregistered(IDisputable(_disputableAddress));
+    }
+
+    /**
+    * @notice Change `_disputable`'s collateral requirements to:
+    * @notice - `@tokenAmount(_collateralToken: address, _actionAmount)` for submitting collateral
+    * @notice - `@tokenAmount(_collateralToken: address, _challengeAmount)` for challenging collateral
+    * @param _disputable Disputable app
+    * @param _collateralToken Address of the ERC20 token to be used for collateral
+    * @param _actionAmount Amount of collateral tokens that will be locked every time an action is submitted
+    * @param _challengeAmount Amount of collateral tokens that will be locked every time an action is challenged
+    * @param _challengeDuration Challenge duration in seconds, during which the submitter can raise a dispute
+    */
+    function changeCollateralRequirement(
+        IDisputable _disputable,
+        ERC20 _collateralToken,
+        uint256 _actionAmount,
+        uint256 _challengeAmount,
+        uint64 _challengeDuration
+    )
+        external
+        auth(MANAGE_DISPUTABLE_ROLE)
+    {
+        DisputableInfo storage disputableInfo = disputableInfos[address(_disputable)];
+        _ensureRegisteredDisputable(disputableInfo);
+
+        _changeCollateralRequirement(_disputable, disputableInfo, _collateralToken, _actionAmount, _challengeAmount, _challengeDuration);
+    }
+
+    /**
+    * @notice Update Agreement to title "`_title`" and content "`_content`", with arbitrator `_arbitrator`
+    * @param _title String indicating a short description
+    * @param _content Link to a human-readable text that describes the rules for the Agreements instance
+    * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
+    */
+    function changeSetting(string _title, bytes _content, IArbitrator _arbitrator) external auth(CHANGE_AGREEMENT_ROLE) {
+        _newSetting(_title, _content, _arbitrator);
+    }
+
+    /**
+    * @notice Sign the agreement
+    */
+    function sign() external {
+        uint256 currentSettingId = _getCurrentSettingId();
+        uint256 lastSettingIdSigned = lastSettingSignedBy[msg.sender];
+        require(lastSettingIdSigned < currentSettingId, ERROR_SIGNER_ALREADY_SIGNED);
```

Comment:
So I think this currently protects this function from being used before being initialized, but we may either want to note that down or add a more explicit check.